core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -661,6 +662,30 @@ private DefaultFunctionResolver date_format() {
     );
   }
 
+  /**
+   * Formats date according to format specifier. First argument is time, second is format.
+   * Detailed supported signatures:
+   * (STRING, STRING) -> STRING
+   * (DATE, STRING) -> STRING
+   * (DATETIME, STRING) -> STRING
+   * (TIME, STRING) -> STRING
+   * (TIMESTAMP, STRING) -> STRING
+   */
+  private DefaultFunctionResolver time_format() {
+    return define(BuiltinFunctionName.TIME_FORMAT.getName(),
+        impl(nullMissingHandling(DateTimeFormatterUtil::getFormattedTime),
+            STRING, STRING, STRING),
```

Comment:
I think you can remove this signature, because smart STRING -> datetime conversion would be added in scope of another task.

Reply:
Left this alone for now as there are tests I wrote which test for this functionality. I can both the signature and tests though if you feel like they should be removed.